autotest/gcore/test_driver_metadata.py

```diff
@@ -64,7 +64,7 @@
           </xs:restriction>
         </xs:simpleType>
       </xs:attribute>
-      <xs:attribute type="xs:string" name="type" use="optional"/>
+      <xs:attribute type="xs:string" name="type" use="required"/>
```

Comment:
could we have some validation of the type with an enumeration ? The accepted ones are listed starting from https://github.com/OSGeo/gdal/blob/master/gdal/gcore/gdaldriver.cpp#L1841

Reply:
I assume the type values are also case insensitive. I did not found a way to check case sensitive with XML schema. Thus I just include lowercase and uppercase variants.